server/jetstream_api.go

```diff
@@ -163,167 +163,213 @@ type ApiError struct {
 	Description string `json:"description,omitempty"`
 }
 
+// ApiResponse is a standard response from the JetStream JSON API
+type ApiResponse struct {
+	Type  string    `json:"type"`
+	Error *ApiError `json:"error,omitempty"`
+}
+
+// ApiPagedResponse includes variables used to create paged responses from the JSON API
+type ApiPagedResponse struct {
```

Comment:
If we call it a response, maybe include `ApiResponse` so that you don't have to include both in real responses? Of call it `ApiPage` maybe.

Reply:
Including it in ApiResponse gets you:

```
var resp Thing{ApiPagedResponse: ApiPagedResponse{ApiResponse: ApiResponse{Type: "x"}}
```

:)

will rename

Reply:
Ouch, your way better then unless we get sane helpers.